v3iofs/fs.py

```diff
@@ -211,39 +214,58 @@ def container_info(container):
     }
 
 
-_missing = object()
-_extra_obj_attrs = [
-    # dest, src, convert
-    ('created', 'creating_time', parse_time),
-    ('atime', 'access_time', parse_time),
-    ('mode', 'mode', lambda v: int(v[1:], base=8)),  # '040755'
-    ('gid', 'gid', lambda v: int(v, 16)),
-    ('uid', 'uid', lambda v: int(v, 16)),
-]
+def prefix_path(container_name, prefix):
+    if not isinstance(prefix, str):
+        prefix = prefix.prefix
+    # prefix may already have a leading / - check and add
+    # if not present
+    if not prefix.startswith("/"):
+        prefix = f'/{prefix}'
+    return unslash(f'/{container_name}{prefix}')
+
+
+def prefix_info(container_name, prefix):
+    return info_of(container_name, prefix, 'prefix')
+
 
+def object_path(container_name, object):
+    # object.key already have a leading /
+    return f'/{container_name}{object.key}'
 
-def obj_path(container, obj, name_key):
-    path = unslash(getattr(obj, name_key))
-    return f'/{container}/{path}'
+
+def object_info(container_name, object):
+    return info_of(container_name, object, 'key')
 
 
 def info_of(container_name, obj, name_key):
     info = {
-        'name': obj_path(container_name, obj, name_key),
-        'type': 'directory' if hasattr(obj, 'size') else 'file',
```

Comment:
Where did `type` go?

Reply:
It is set in self._details_of, which handles defining files and directories